### DISCUSSION TITLE
Advance thermal disinfection day when disinfection temperature is reached by solar

### DISCUSSION THREAD

**Discussion by lievendf** [emsesp/EMS-ESP32#1768] [Ideas]
Hi,

My Buderus installation is configured to heat the dhw to 70 degrees once a week at night (1 am) to prevent bacteria growth (Legionella).
In summer it frequently happens that the dhw temperature reaches 70 degrees or higher purely by solar power.
When that happens it would be beneficial to advance the configured "dhw disinfection day" to the day this high temperature is reached, effectively postponing the planned disinfection for one week.

I'm running Home Assistant, and there might be a way to program an automation for this, but I thought of asking here first as people might have come up with a solution for this.
Alternatively a built-in solution in ems-esp would be great for all users that have a solar module.

Note that lower temperatures could be valid as well, as long as they are maintained for a longer time:
> If water is maintained at 50°C for two hours, 90% of the bacteria will be killed off. If water reaches 60°C, the same percentage of bacteria will be killed off in two minutes. If the water reaches 70°C, all the legionella bacteria will die instantly. https://www.urisk.co.uk/what-temperature-kills-legionella

Thanks,
Lieven

**Comment by proddy** [comment]
I'm guessing your thermostat triggers the automatic disinfection based on day/time. The simplest way could be to use HA to trigger manual disinfection (by sending the command to EMS-ESP) based on the actual current dhw temp at 1am on a specific day.

I'm always looking for new features to add to EMS-ESP.  There are many use cases where we should really extend the Scheduler with simple conditions (if..then..else).

**Reply by lievendf** [reply]
Yes, it's the RC35 that has "dhw disinfecting", "dhw disinfection day" and "dhw disinfection hour".
"dhw disinfection temperature" is part of the boiler settings (GB162).

~~There doesn't appear to be a setting to trigger an immediate disinfection~~ There appears to be a "dhw disinfecting" setting on the boiler (GB162) as well, which appears to be "off" currently, while the same named setting on the thermostat (RC35) appears to be on. Could the one on GB162 be able to start an immediate disinfection?

Before finding the above, I've started looking at making a HA automation to advance the day when the temperature reaches a specific threshold.
I've created a template sensor "sensor.ems_esp_day_of_week" with the following code:
```
{% set weekdays = ['mo', 'tu', 'we', 'th', 'fr', 'sa', 'su'] %}
{% set day_number = now().weekday() %}
{{ weekdays[day_number] }}
```
Then I've created a new numeric_state automation:
```
platform: numeric_state
entity_id:
  - sensor.boiler_wwcurtemp
above: 69
service: select.select_option
target:
  entity_id:
    - select.thermostat_wwdisinfectday
data:
  option: "{{ states('sensor.ems_esp_day_of_week') }}"
```

I might add variants to this one using lower temperatures combined with a longer duration.
Now I just have to wait for a sunny day to verify if everything works...
